### PR TITLE
fix: resolved ts error "instana.currentSpan is not a type"

### PR DESCRIPTION
**Pull request by kirrg001**
Add a couple of more typescript types attributes.
Added more tests for typescript (TS->CJS, TS->ESM).
Removed dist folder from git as discussed.

### CODE REVIEW

File: packages/collector/src/types/shared.d.ts [hunk 1]
```diff
@@ -17,6 +17,10 @@ export type InitFunction = {
   (config?: CollectorConfig): Init;
   sdk: any;
   core: any;
+  currentSpan: any;
+  isTracing: any;
+  isConnected: any;
+  setLogger: any;
```
Comment (kirrg001) [inline packages/collector/src/types/shared.d.ts:23]: This is the fix.

File: packages/collector/test/tracing/misc/node-fetch/app.js [hunk 2]
```diff
@@ -39,8 +39,8 @@ app.get('/', async (req, res) => {
 });
 
 app.get('/request', async (req, res) => {
+  log('Received request /request');
```
Comment (aryamohanan) [inline packages/collector/test/tracing/misc/node-fetch/app.js:42]: This log required?

Comment (abhilash-sivan) [inline packages/collector/test/tracing/misc/node-fetch/app.js:42]: Do we need this ?

File: packages/collector/test/tracing/misc/typescript/ts_cjs/ts_cjs_test.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    before(async () => {
+      execSync('rm -rf ./node_modules', { cwd: __dirname, stdio: 'inherit' });
+      execSync('npm install --verbose', { cwd: __dirname, stdio: 'inherit' });
+      execSync('npm run build', { cwd: __dirname, stdio: 'inherit' });
+
+      controls = new ProcessControls({
+        appPath: path.join(__dirname, 'dist', 'app_1.js'),
+        useGlobalAgent: true,
+        execArgv: ['--require', path.join(__dirname, 'node_modules', '@instana', 'collector', 'src', 'immediate.js')]
```
Comment (abhilash-sivan) [inline packages/collector/test/tracing/misc/typescript/ts_cjs/ts_cjs_test.js:36]: 👍

File: packages/collector/test/tracing/misc/typescript/ts_esm/ts_esm_test.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      execSync('rm -rf ./dist', { cwd: __dirname, stdio: 'inherit' });
+      execSync('rm -rf ./node_modules', { cwd: __dirname, stdio: 'inherit' });
+      execSync('npm install --verbose', { cwd: __dirname, stdio: 'inherit' });
+      execSync('npm run build', { cwd: __dirname, stdio: 'inherit' });
+      execSync('cp ./package.json.tmp ./dist/package.json', { cwd: __dirname, stdio: 'inherit' });
+
+      controls = new ProcessControls({
+        appPath: path.join(__dirname, 'dist', 'app_1.js'),
+        useGlobalAgent: true,
+        execArgv: ['--import', path.join(__dirname, 'node_modules', '@instana', 'collector', 'esm-register.mjs')]
```
Comment (aryamohanan) [inline packages/collector/test/tracing/misc/typescript/ts_esm/ts_esm_test.js:38]: What will be happen if we run this in lower nodejs versions? The command with--import work only for 18.19 and above?

File: packages/collector/test/tracing/misc/typescript/ts_esm/ts_esm_test.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      execSync('rm -rf ./dist', { cwd: __dirname, stdio: 'inherit' });
+      execSync('rm -rf ./node_modules', { cwd: __dirname, stdio: 'inherit' });
+      execSync('npm install --verbose', { cwd: __dirname, stdio: 'inherit' });
+      execSync('npm run build', { cwd: __dirname, stdio: 'inherit' });
+      execSync('cp ./package.json.tmp ./dist/package.json', { cwd: __dirname, stdio: 'inherit' });
+
+      controls = new ProcessControls({
+        appPath: path.join(__dirname, 'dist', 'app_2.js'),
+        useGlobalAgent: true,
+        execArgv: ['--import', path.join(__dirname, 'node_modules', '@instana', 'collector', 'esm-register.mjs')]
```
Comment (aryamohanan) [inline packages/collector/test/tracing/misc/typescript/ts_esm/ts_esm_test.js:81]: same here?

File: packages/collector/test/tracing/misc/node-fetch/app.js (hunk 2 above)
Comment (kirrg001) [reply packages/collector/test/tracing/misc/node-fetch/app.js:42]: Yes it gives visibility while running the tests.

Comment (kirrg001) [reply packages/collector/test/tracing/misc/node-fetch/app.js:42]: I added this in the wrong app. Thanks

Comment (kirrg001) [reply packages/collector/test/tracing/misc/node-fetch/app.js:42]: Reverting.

File: packages/collector/test/tracing/misc/typescript/ts_esm/ts_esm_test.js (hunk 4 above)
Comment (kirrg001) [reply packages/collector/test/tracing/misc/typescript/ts_esm/ts_esm_test.js:38]: Yes 👍 Will add the conditional check thanks

File: packages/collector/test/tracing/misc/typescript/ts_esm/ts_esm_test.js (hunk 5 above)
Comment (kirrg001) [reply packages/collector/test/tracing/misc/typescript/ts_esm/ts_esm_test.js]: Changed

File: packages/collector/test/tracing/misc/typescript/ts_esm/ts_esm_test.js (hunk 4 above)
Comment (kirrg001) [reply packages/collector/test/tracing/misc/typescript/ts_esm/ts_esm_test.js]: Changed

File: packages/collector/test/tracing/misc/node-fetch/app.js (hunk 2 above)
Comment (kirrg001) [reply packages/collector/test/tracing/misc/node-fetch/app.js]: Reverted

**Review by abhilash-sivan — APPROVED** [review]

**Review by aryamohanan — APPROVED** [review]
Thanks for adding the tests ❤️

**Pull request merged**